app-gateway-staging.tf

```diff
@@ -0,0 +1,96 @@
+data "azurerm_key_vault" "infra_vault" {
+  name                = "infra-vault-${var.subscription}"
+  resource_group_name = "${var.env == "prod" ? "core-infra-prod" : "cnp-core-infra"}"
+}
```

Comment:
it is already defined in `app-gateway.tf`

Reply:
Cleaned up the resources/duplicate usage now